### PR TITLE
Feature/co 468 subjects

### PR DESCRIPTION
**Pull request by floppyears**

### CODE REVIEW

File: src/main/groovy/edu/oregonstate/mist/api/BuildInfoManager.groovy [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ */
+class BuildInfoManager implements Managed {
+    Info getInfo() {
+        info
+    }
+    private Info info = new Info()
+
+    @Override
+    public void start() throws Exception {
+        def stream = InfoResource.class.getResourceAsStream('/build.properties')
```
Comment (SuzyWu2014) [inline src/main/groovy/edu/oregonstate/mist/api/BuildInfoManager.groovy:16]: If I understand it correctly, this method is meant to populate info when the application starts. Then why do you call getResourceAsStream from InfoResource class? would it make any difference if you call it from Resource or itself if possible?

File: src/main/groovy/edu/oregonstate/mist/api/Resource.groovy [hunk 2]
```diff
@@ -1,14 +1,35 @@
...
+
+    @Context
+    UriInfo uriInfo
+
+    /**
+     * URI used to provide jsonapi pagination links.
+     */
+    private URI endpointUri
+
     public static loadProperties() {
```
Comment (SuzyWu2014) [inline src/main/groovy/edu/oregonstate/mist/api/Resource.groovy:33]: why not  "public static loadProperties() throws Exception" if the exception is not handled here? Something special about static methods?

File: src/main/groovy/edu/oregonstate/mist/api/BuildInfoManager.groovy (hunk 1 above)
Comment (floppyears) [inline src/main/groovy/edu/oregonstate/mist/api/BuildInfoManager.groovy:16]: That was the case in a previous commit. In the latest commit, that's not the case. Would you mind taking a look at these:
https://github.com/osu-mist/course-subjects-api/blob/feature/CO-468-subjects/src/main/groovy/edu/oregonstate/mist/api/InfoResource.groovy
https://github.com/osu-mist/course-subjects-api/blob/feature/CO-468-subjects/src/main/groovy/edu/oregonstate/mist/api/BuildInfoManager.groovy

File: src/main/groovy/edu/oregonstate/mist/api/Resource.groovy (hunk 2 above)
Comment (floppyears) [inline src/main/groovy/edu/oregonstate/mist/api/Resource.groovy:33]: Good catch. Would you mind creating a ticket for this? The code in this package is part of the skeleton, so we should handle it in that repo.

File: src/main/groovy/edu/oregonstate/mist/api/BuildInfoManager.groovy (hunk 1 above)
Comment (SuzyWu2014) [inline src/main/groovy/edu/oregonstate/mist/api/BuildInfoManager.groovy:16]: Okay, I see the changes, but can we do 
def stream = this.class.getResourceAsStream('/build.properties') in BuildInfoManager.groovy, I wonder why you call it from InfoResource when you are at BuildInfoManager?

**Pull request merged**